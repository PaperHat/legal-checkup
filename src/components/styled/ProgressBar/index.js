import styled from 'styled-components'

const ProgressBar = styled.ul`
  list-style: none;
  display: flex;
  background-color: gainsboro;
  margin: 0 0.2rem;
  padding: 0;
`

ProgressBar.displayName = 'StyledProgressBar'

export default ProgressBar
