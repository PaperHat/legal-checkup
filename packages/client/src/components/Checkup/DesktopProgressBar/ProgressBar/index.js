import styled from 'styled-components'

const ProgressBar = styled.nav`
  background-color: #fff;
  display: flex;
  justify-content: center;
  margin: 20px 0;
`

ProgressBar.displayName = 'StyledDesktopProgressBar'

export default ProgressBar
