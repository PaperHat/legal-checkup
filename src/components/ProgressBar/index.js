import React from 'react'
import styled from 'styled-components'

const bar = styled.bar`
width: ${props => props.progress}  
height: 10px;
background-color: #4CAF50;
`
export default class ProgressBar extends React.Component {
  constructor (props) {
    super(props)
    this.state = {
      progress: 0,
      question_number: ''
    }
    this.checkProgress = this.checkProgress.bind(this)
    this.renderProgressBar = this.renderProgressBar.bind(this)
  }

  checkProgress (question_number) {
    let progress = 0

    if (question_number > 0) {
      return (progress += 1)
    }
  }

  render () {

  }
}
