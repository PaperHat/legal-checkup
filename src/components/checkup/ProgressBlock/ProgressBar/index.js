import styled from 'styled-components'
import styledMap from 'styled-map'

import { DESKTOP, MOBILE, TABLET } from '../../../../constants'

const ProgressBar = styled.nav`
  background-color: #eeedea;
  display: flex;
  width: 100%;
  flex-wrap: nowrap;
  align-items: center;
  justify-content: center;
  letter-spacing: normal;
  text-align: center;
  margin: 20px 0;

  ${styledMap('format', {
    [DESKTOP]: ``,
    [MOBILE]: ``,
    [TABLET]: ``
  })}
`

ProgressBar.displayName = 'ProgressBar'

export default ProgressBar
