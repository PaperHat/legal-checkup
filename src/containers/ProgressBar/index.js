import { connect } from 'react-redux'

import { ProgressBar } from '../../components'
import {
  getActiveQuestion,
  getQuestions,
  getResponses,
  questionSelected,
  previousQuestion,
  nextQuestiion
} from '../../state'

function mapStateToProps (state) {
  return {
    activeQuestion: getActiveQuestion(state),
    questions: getQuestions(state),
    responses: getResponses(state)
  }
}

function mapDispatchToProps (dispatch) {
  return {
    handleClick: questionNumber => dispatch(questionSelected(questionNumber)),
    handlePrevious: () => dispatch(previousQuestion()),
    handleNext: () => dispatch(nextQuestiion())
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(ProgressBar)
