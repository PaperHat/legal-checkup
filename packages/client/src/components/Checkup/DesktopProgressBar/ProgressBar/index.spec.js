import React from 'react'

import { shallow } from 'enzyme'

import ProgressBar from './'

describe('components:styled:ProgressBar', () => {
  it('renders the ProgressBar and CSS properties properly', () => {
    expect(toJson(shallow(<ProgressBar />))).toMatchSnapshot()
  })
})
