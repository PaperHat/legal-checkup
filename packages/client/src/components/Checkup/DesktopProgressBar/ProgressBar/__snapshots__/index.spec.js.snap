// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components:styled:ProgressBar renders the ProgressBar and CSS properties properly 1`] = `
<StyledComponent
  forwardedClass={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": undefined,
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
  background-color: #fff;
  display: flex;
  justify-content: center;
  margin: 20px 0;
",
        ],
      },
      "displayName": "DesktopProgressBar",
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "nav",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
